components/ConfirmationModal.js

```diff
@@ -41,18 +41,18 @@ export const CONFIRMATION_MODAL_TERMINATE = { __CONFIRMATION_MODAL_TERMINATE: tr
  * confirmation purpose.
  */
 const ConfirmationModal = ({
-  header,
-  body,
-  children,
+  header = undefined,
+  body = undefined,
+  children = undefined,
   onClose,
-  type,
-  isDanger,
-  isSuccess,
-  cancelLabel,
-  continueLabel,
-  cancelHandler,
+  type = 'confirm',
+  isDanger = undefined,
+  isSuccess = undefined,
+  cancelLabel = undefined,
+  continueLabel = undefined,
+  cancelHandler = onClose,
   continueHandler,
-  disableSubmit,
+  disableSubmit = undefined,
```

Comment:
Is there a reason for explicitly setting the default to `undefined`? - if not, seems to make it harder to see what fields have a default set that is not the implicit default which is already `undefined`

Reply:
Typesccript will consider `myProp = undefined` as an optional prop (because it has a default value) and thus don't error because it's not set. Doing this is a cheap way to make a component compatible with typescript without digging too much.